### PR TITLE
admin: Add support for comment search by Thread URL in admin interface

### PR DESCRIPTION
**Pull request by pkvach**
## Checklist
- [x] All new and existing **tests are passing**
- [x] I have added an entry to `CHANGES.rst` because this is a user-facing change or an important bugfix
- [x] I have written **proper commit message(s)**
      

## What changes does this Pull Request introduce?

The main theme of the changes is to enhance the comment search functionality in the admin interface. The changes allow searching for comments not only by comment URL but also by thread URL.

## Why is this necessary?

- https://github.com/isso-comments/isso/issues/642
- https://github.com/isso-comments/isso/pull/1000

### CODE REVIEW

**Review by ix5 — CHANGES_REQUESTED** [review]
Thanks for the PR!

Could we perhaps find a better way to convey to admins that they can enter both thread and individual comment URLs?

Also, would it make sense to make the search field a bit smarter and just let it accept plain numbers (for comment ids) as well?

**Comment by pkvach** [comment]
> Thanks for the PR!

I'm happy to help :)

> Could we perhaps find a better way to convey to admins that they can enter both thread and individual comment URLs?

There is an option to add a tooltip with explanatory text. This will not take up too much space.

> Also, would it make sense to make the search field a bit smarter and just let it accept plain numbers (for comment ids) as well?

Then we will definitely need explanatory text and will have to give up `<input type=“url”>`.

**Comment by ix5** [comment]
> > Could we perhaps find a better way to convey to admins that they can enter both thread and individual comment URLs?
> 
> There is an option to add a tooltip with explanatory text. This will not take up too much space.

Good idea. Something like `<abbr>` that pops on mouseover will not work on mobile, but if you have a good solution, go for it.

> > Also, would it make sense to make the search field a bit smarter and just let it accept plain numbers (for comment ids) as well?
> 
> Then we will definitely need explanatory text and will have to give up `<input type=“url”>`.

On second thought, let's not implement the "smart" search and leave it as is. It would just be confusing.

**Comment by pkvach** [comment]
I added a tooltip with help text next to the search field.
It's ready for review.

File: isso/css/admin.css [hunk 1]
```diff
@@ -176,3 +176,67 @@ a {
...
+
+.search-tooltip .show {
+  visibility: visible;
+  -webkit-animation: fadeIn 1s;
+  animation: fadeIn 1s;
+}
+
+/* Add animation (fade in the tooltip) */
+@-webkit-keyframes fadeIn {
+  from {opacity: 0;}
```
Comment (ix5) [inline isso/css/admin.css:235]: single trailing whitespace here
```diff
-  from {opacity: 0;} 
+  from {opacity: 0;}
```

**Review by ix5 — COMMENTED** [review]
Nice, the tooltip is really useful.

I'd just want it a bit more integrated/look alike the rest of the surrounding buttons, so that it's clearer it is clickable.

Current:
Screenshot from 2024-05-03 19-03-18

With more prominent border and shadow:
Screenshot from 2024-05-03 19-02-55

**Comment by pkvach** [comment]
> Nice, the tooltip is really useful.
> 
> I'd just want it a bit more integrated/look alike the rest of the surrounding buttons, so that it's clearer it is clickable.
> 
> Current: Screenshot from 2024-05-03 19-03-18
> 
> With more prominent border and shadow: Screenshot from 2024-05-03 19-02-55

Thank you for the review and a nice addition.
I made the suggested changes and did squash commits.

**Review by ix5 — APPROVED** [review]

**Comment by ix5** [comment]
Merged, thanks!

**Pull request merged**